docs/design/distributed-framework-plugins.md

```diff
@@ -199,6 +193,108 @@ func (tp *tensorflowPlugin) getClusterInfo(job *batch.Job) clusterInfo {
 }
 ```
 
+#### Pytorch Plugin
+
+Similar to the tensorflow plugin, firstly we must know the cluster role of task in volcano job, and the port to be exposed. And this information can be passed by plugin arguments, which is defined in job spec.
+
+```yaml
+spec:
+  plugins:
+    # set pytorch plugin
+    pytorch: ["--master=master","--worker=worker","--port=23456"]
+```
+
+In the implementation of `pytorchPlugin`, these arguments will be parsed.
+
+```go
+// pytorchPlugin is plugin for pytorch framework
+type pytorchPlugin struct {
+	pytorchArguments []string
+	clientset        pluginsinterface.PluginClientset
+	masterName       string
+	workerName       string
+	port             int
+}
+// parse all arguments
+func (pp *pytorchPlugin) addFlags() {
+	flagSet := flag.NewFlagSet(pp.Name(), flag.ContinueOnError)
+	flagSet.StringVar(&pp.masterName, "master", DefaultMaster, "name of master role task")
+	flagSet.StringVar(&pp.workerName, "worker", DefaultWorker, "name of worker role task")
+	flagSet.IntVar(&pp.port, "port", DefaultPort, "open port for containers")
+	if err := flagSet.Parse(pp.pytorchArguments); err != nil {
+		klog.Errorf("plugin %s flagset parse failed, err: %v", pp.Name(), err)
+	}
+}
+```
+
+Then we patch pytorch-distributed-training related environment variables to container envs in method `OnPodCreate`.
+The main environment variables are:
+* `MASTER_ADDR`: master address
+* `MASTER_PORT`: master port
+* `WORLD_SIZE`: total node number
+* `RANK`: current node index
+
+```go
+func (pp *pytorchPlugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
```

Comment:
Same advice here. Please remove the implementation from the design doc.

Reply:
ok